### ISSUE TITLE
Cap3

### ISSUE THREAD

**Issue by leehambley** [crowdfavorite/gem-steps#9]
I've just had a mailing list question for Capistrano regarding the integration. Looks like your Gem is targeting the 2.x code branch which was EOL'ed recently.

There's a new rewrite, and I'm not sure how you can integrate, if you are even interested in doing so.

I just wanted to make you aware of the issue, I've cross-posted this issue reference to the list's OP, too:
- Mailing list thread: https://groups.google.com/forum/#!topic/capistrano/bdZ_17b32fI

Thanks!

**Comment by jondavidjohn** [comment]
Cool, thanks for the heads up!

The gem itself isn't really "targeted" at a particular version of capistrano, we just expose some one off methods to control the output.

So really the question is how would someone control the verbosity of the output similar to the 2.x method of

``` ruby
      logger.level = Logger::IMPORTANT      # or Capistrano::Logger::IMPORTANT
```

And then what does the before/after syntax look like compared to the integration example I give in the README.md

**Comment by jondavidjohn** [comment]
@leehambley Also, in the mailing list thread you mention that the `pretty` formatter is swappable.  Could you point me in that direction if that seems like a cleaner point of integration.

**Comment by leehambley** [comment]
The `Pretty` formatter is here: https://github.com/leehambley/sshkit - it could be modified to accept other types of objects, or replaced with something less verbose (we prefer verbosity as it equals transparency for new users) - It's tested (not comprehensively, but it's there)

If you want to target leaner logging solutions to Cap it's worth noting according to the new DSL (http://capistranorb.com) anything in the `on()` block is evaluated inside of SSHKit's loaded backend, which means some glue code if you want parity inside and outside of `on`, but still with in task (we solved this wt a logging modeule injected into both places)

If you do hack something up, you'll be the first consumer of the new IO orientated printer strategy, so if there's any changes we can make to support you, feel free to let us know.

**Comment by jondavidjohn** [comment]
@leehambley 

Looked in to this briefly, and if I understand you correctly Cap 3 output is generated from multiple places, even pieces that are outside of Capistrano? via SSHKit?

Seems like this would be difficult to customize.

You mentioned an IO oriented printer, is there a way to provide your own printer?

**Comment by leehambley** [comment]
Yes, just provide the class and provide it to SSJKit as the printer io

**Comment by ddarbyson** [comment]
+1 for Cap3 compatibility

**Comment by Aupajo** [comment]
:+1:

**Comment by jeffbyrnes** [comment]
:+1:

**Comment by leehambley** [comment]
https://github.com/capistrano/sshkit/blob/master/EXAMPLES.md#implement-a-dirt-simple-formatter-class
